### ISSUE TITLE
[a11y]: Full Page Error, Screen Reader Incorrectly Announces "Open Menu" for Close Icon

### ISSUE THREAD

**Issue by anamikaanu96** [carbon-design-system/ibm-products#6230]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.52.0-rc.0

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during QA Audit.

After activating the open menu, the close icon appears, but the screen reader announces it as "open menu" instead of providing the correct context for the close action.

The close icon should be announced accurately by the screen reader as "close menu" or similar, clearly indicating its function to users. This ensures that users understand that they can close the menu rather than opening it again.

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-full-page-error-fullpageerror--full-page-error&globals=viewport:basic

### Steps to reproduce

This was reported in all variants of the Full Page Error.

Open Full Page Error, Click on the hamburger menu and observe screen Reader Announcement - incorrectly announcing  "Open Menu" for Close Icon

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
